src/app/Joyride/JoyrideProvider.tsx

```diff
@@ -35,46 +35,54 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import { LoadingView } from '@app/LoadingView/LoadingView';
-import {
-  QuickStartCatalogPage,
-  QuickStartContainer,
-  QuickStartContainerProps,
-  useLocalStorage,
-} from '@patternfly/quickstarts';
+import useSetState from '@app/utils/useSetState';
 import React from 'react';
-import { useTranslation } from 'react-i18next';
-import { withRouter } from 'react-router-dom';
-import { allQuickStarts } from './all-quickstarts';
+import { Step } from 'react-joyride';
 
-export interface QuickStartsProps {}
+export interface JoyrideState {
+  run: boolean;
+  stepIndex: number;
+  steps: Step[];
+}
 
-const QuickStarts: React.FunctionComponent<QuickStartsProps> = (_) => {
-  const { t, i18n } = useTranslation();
-  const [activeQuickStartID, setActiveQuickStartID] = useLocalStorage('quickstartId', '');
-  const [allQuickStartStates, setAllQuickStartStates] = useLocalStorage('quickstarts', {});
+const defaultState = {
+  run: false,
+  stepIndex: 0,
+  steps: [] as Step[],
+};
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+export const JoyrideContext = React.createContext({
+  state: defaultState,
+  setState: (_patch: Partial<JoyrideState> | ((previousState: JoyrideState) => Partial<JoyrideState>)) => {},
+  setIsNavBarOpen: (_isOpen: boolean) => {},
+  isNavBarOpen: true,
+});
+/* eslint-enable @typescript-eslint/no-empty-function */
 
-  const drawerProps: QuickStartContainerProps = {
-    quickStarts: allQuickStarts,
-    activeQuickStartID,
-    allQuickStartStates,
-    setActiveQuickStartID,
-    setAllQuickStartStates,
-    language: i18n.language,
-    alwaysShowTaskReview: true,
-  };
+export const JoyrideProvider: React.FC<{ children }> = (props) => {
+  const [state, setState] = useSetState(defaultState);
+  const [isNavBarOpen, setIsNavBarOpen] = React.useState(true);
+  const value = React.useMemo(
+    () => ({ state, setState, isNavBarOpen, setIsNavBarOpen }),
+    [state, setState, isNavBarOpen, setIsNavBarOpen]
+  );
   return (
-    <React.Suspense fallback={<LoadingView />}>
-      <QuickStartContainer {...drawerProps}>
-        <QuickStartCatalogPage
-          title={t('QuickStarts.CATALOG_PAGE.TITLE')}
-          hint={t('QuickStarts.CATALOG_PAGE.HINT')}
-          showTitle
-          showFilter
-        />
-      </QuickStartContainer>
-    </React.Suspense>
+    <JoyrideContext.Provider value={value} {...props}>
+      {props.children}
+    </JoyrideContext.Provider>
   );
 };
 
-export default withRouter(QuickStarts);
+export const useJoyride = (): {
+  setState: (patch: Partial<JoyrideState> | ((previousState: JoyrideState) => Partial<JoyrideState>)) => void;
+  state: JoyrideState;
+  isNavBarOpen: boolean;
+  setIsNavBarOpen: React.Dispatch<React.SetStateAction<boolean>>;
+} => {
+  const context = React.useContext(JoyrideContext);
+  if (context === undefined) {
```

Comment:
This might not ever be reached since if no provider is found for this context, the default value is returned.